modules/lib/src/main/resources/assets/js/app/inputtype/ui/text/dialog/LinkModalDialog.ts

```diff
@@ -657,7 +657,13 @@ export class LinkModalDialog
 
         this.paramsFormIds.push({keyId: keyFormItemId, valueId: valueFormItemId});
 
-        keyFormItem.getInput().giveFocus();
+        const keyFormItemInput = <TextInput>keyFormItem.getInput();
+
+        keyFormItemInput.onValueChanged(() => {
+            if (this.validationQueryParams.bind(this)() === undefined) {
+                this.paramsFormItem.validate(new ValidationResult(), true);
+            }
```

Comment:
It feels like a strange way to solve this... You are checking if validation error message is undefined, and in this case you are running validation. 

Maybe it's the architecture of these methods that is wrong originally. It should be something like this:

```
private isQueryParamsValid(): boolean {
  const isValid = Array.from(this.paramsFormItem.getHTMLElement().getElementsByTagName('input'))
  .filter((input: HTMLInputElement) => input.className.indexOf('params-key') >= 0)
  .every((input: HTMLInputElement) => input.value.trim() !== '');

  return isValid;
}

private validationQueryParams(): string {
  return !this.isQueryParamsValid() ? i18n('dialog.link.queryparams.empty') : undefined;
}
```

Then in your listener you could do this:
```
keyFormItemInput.onValueChanged(() => {
  if (!this.isQueryParamsValid()) {
    this.paramsFormItem.validate(new ValidationResult(), true);
  }
});
```